File: backend/src/main/java/org/example/backend/modules/edu/entity/StageSubmission.java
package org.example.backend.modules.edu.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("stage_submission")
public class StageSubmission {
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private Long taskId;

    private Long batchId;

    private Long groupId;

    private Long submitterId;

    private Integer versionNo;

    private String summary;

    private String reportText;

    private String repoUrl;

    private String deployUrl;

    private Integer status;

    private LocalDateTime submitTime;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/stageSubmission.ts
import request from '@/utils/request'

export interface StageSubmissionItem {
  id: string
  taskId: string
  batchId: string
  groupId: string
  submitterId: string
  versionNo: number
  summary: string | null
  reportText: string | null
  repoUrl: string | null
  deployUrl: string | null
  status: number
  submitTime: string | null
  createTime: string | null
  updateTime: string | null
}

export interface StageSubmissionPageQuery {
  current: number
  size: number
  versionNo?: number
  summary?: string
  status?: number
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export interface StageSubmissionFormData {
  id?: string
  taskId: string
  batchId: string
  groupId: string
  versionNo: number
  summary: string
  reportText: string
  repoUrl: string
  deployUrl: string
  status: number
}

export const getStageSubmissionPageApi = (params: StageSubmissionPageQuery) => {
  return request.get<any, { code: number; message: string; data: PageResult<StageSubmissionItem> }>('/api/stage-submissions/page', {
    params,
  })
}

export const createStageSubmissionApi = (data: StageSubmissionFormData) => {
  return request.post<any, { code: number; message: string; data: StageSubmissionItem }>('/api/stage-submissions', data)
}

export const updateStageSubmissionApi = (data: StageSubmissionFormData) => {
  return request.put<any, { code: number; message: string; data: StageSubmissionItem }>('/api/stage-submissions', data)
}

export const getStageSubmissionDetailApi = (id: string) => {
  return request.get<any, { code: number; message: string; data: StageSubmissionItem }>(`/api/stage-submissions/${id}`)
}

export const deleteStageSubmissionApi = (id: string) => {
  return request.delete<any, { code: number; message: string; data: boolean }>(`/api/stage-submissions/${id}`)
}

export const getStageSubmissionListApi = () => {
  return request.get<any, { code: number; message: string; data: StageSubmissionItem[] }>('/api/stage-submissions')
}